### ISSUE TITLE
Use Geoclue for geolocation

### ISSUE THREAD

**Issue by hadess** [rhinstaller/anaconda#2]
Instead of custom server and code.

**Comment by vpodzime** [comment]
We might need to add some code to Geoclue so that it uses our "custom" server because we need to use a Fedora-intra/infra server instead of any third-party server.

**Comment by M4rtinK** [comment]
Also the Python bindings for Geoclue seem to be Python 2 only:
https://admin.fedoraproject.org/pkgdb/package/python-geoclue/
Actually, it kinda looks like retired as of F21:
http://pkgs.fedoraproject.org/cgit/python-geoclue.git/commit/?h=f21

**Comment by hadess** [comment]
@vpodzime Why? The Mozilla services we use are more precise, and actually have a privacy policy[1]. The server you use right now doesn't.

@M4rtinK Geoclue uses a pure D-Bus interface. The bindings you point to are for the obsolete Geoclue 1.

[1]: https://location.services.mozilla.com/privacy

**Comment by M4rtinK** [comment]
@hadess: Well it claims: "This Python module uses the Geoclue D-Bus API to implement a nice API for python developers and facilitate their life." So there are two Geoclue DBUS APIs ?

BTW,  the Mozilla API is a WiFi location API, right ? That could indeed be used to replace the proof of concept WiFi location code in pyanaconda/geoloc.py.

As for the default location API, the Fedora hosted location API is just doing simple GeoIP based geolocation. This is on purpose, as we don't want to leak such sensitive information as nearby access point addresses by default. Even the GeoIP lookup goes through a Fedora infra hosted server, so that we don't leak the IP address of each interactive Fedora installation to a random third party.

So in short, we should be able to use Geoclue for (optional!) WiFi based geolocation, but unless it can use custom GeoIP providers, we can't use it to replace our custom Fedora GeoIP API lookup code.

**Comment by poncovka** [comment]
Thank you for reporting this issue! The issue tracker will be closed on 3 June 2019. Please make sure that your issue is also reported at https://bugzilla.redhat.com/. We are handling a high number of issues and many of them involve other components. Bugzilla allows us to better process your issues and collaborate on them with other teams. We are sorry for the inconvenience.
